packages/blade/src/components/Box/BaseBox/types/propsTypes.ts

```diff
@@ -15,6 +15,7 @@ type LayoutProps = MakeObjectResponsive<
     width: SpacingValueType;
     minWidth: SpacingValueType;
     maxWidth: SpacingValueType;
+    textAlign: CSSObject['textAlign'];
   } & PickCSSByPlatform<'display' | 'overflow' | 'overflowX' | 'overflowY'>
```

Comment:
```suggestion
  } & PickCSSByPlatform<'display' | 'overflow' | 'overflowX' | 'overflowY' | 'textAlign'>
```

Reply:
It was throwing some error before but not throwing now, maybe some issue was with TSServer. fixed.

Reply:
Okay yeah this errors out on ReactNative. because react-native doesn't have textAlign styling on View props. 

I think it will be better to just keep the types same and ignore the prop on react-native. otherwise we will have to put `textAlign={'left' as never}` on all the places where we used it since it will work on web but fail on native

<img width="867" alt="image" src="https://github.com/razorpay/blade/assets/35374649/9bb1caa7-ef06-4e8c-911f-a97901286acf">

Reply:
`textAlign={castWebType('left')}` should fix it. This is what we do on other props as well.

Reply:
> his is what we do on other props as well.

Oh really? 

That is actually not the correct way to use CastWebType & CastNativeType. 
These cast functions are suppose to only work with `Branded<>` types (eg: types generated by Platform.Select) 

if you do `castWebType('some random type without brand')` this will return `never` 

So you are basically doing the same thing as `textAlign={'left' as never}`

Reply:
no so we are using `Platform.Select`. That PickCSSByPlatform utility is using `Platform.Select` to decide which type to pick on which platform.

Reply:
I think let's do this: 

1. Remove all the wrong usage of castWebType and let's just directly cast them to native since they are either gonna break on web or on native if there is no overlap of the type
2. I'll change the textAlign to `PickCSSByPlatform` so that consumers will atleast get correct types. 
3. I'll also change castWeb & castNative to not accept any random type and only accept Branded types. 

Sounds good?

Reply:
This one is the branded type though no? This is what PickCSSByPlatform does 

<img width="929" alt="image" src="https://github.com/razorpay/blade/assets/30949385/a7a6eb74-7329-42e5-9e7d-7ed4f236e9e0">

Reply:
It does seem to work as expected (added screenshots in this PR) https://github.com/razorpay/blade/pull/1113 

Let me know if I missed something

Reply:
Ohhhhhh understood. You meant we have to explicitly pass `as Type` to castWebType 🤦🏼 got it lol I for some reason thought it magically figures out type in web for that prop but yeah makes sense.

Reply:
<img width="678" alt="image" src="https://github.com/razorpay/blade/assets/30949385/c2bf7317-d44e-4e13-ab0f-4c4b63c754c5">
ahh yess this makes more sense 🤦🏼

Reply:
I think let it be for this PR. Will create separate PR to fix these types.

Reply:
Yeah, let me to this atleast for the textAlign will fix other types later.